assets/prototype/domain/shared/predicates/datetimes/filters/allDates/index.test.ts

```diff
@@ -0,0 +1,26 @@
+import dates from '../tests/data';
+import allDates from './index';
+
+describe('allDates', () => {
+	it('Should return an array with active-only dates', () => {
+		const filteredDates = allDates(dates);
+
+		expect(Array.isArray(filteredDates)).toBe(true);
+		filteredDates.forEach((date) => {
+			// Since in Datetime this is set as optional, we need to check first is this is defined first
+			if (date.isDeleted) {
```

Comment:
Shouldn't it be `if (date.isDeleted === undefined)` to check if `isDeleted` exists?

Reply:
It should be `if (date.isDeleted !== undefined)`, updated.

Reply:
`if (date.isDeleted !== undefined)` is the same as `if (date.isDeleted)`, but added the more explicit version.